### PR TITLE
Implemented modules versions

### PR DESCRIPTION
**Pull request by aleksandrychev**
Ticket: ENT-7736

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -42,6 +46,9 @@ module.exports = function (grunt) {
 
         let processFile = function (abspath, filename) {
             let content = grunt.file.read(abspath);
+            if (!content.includes('commit')) {
+                return null
+            }
             let pageIndex;
             // First separate the Front Matter from the content and parse it
             content = content.split("}     \n");
```
Comment (craigcomstock) [inline Gruntfile.js:54]: Why do we split on such an odd string? `}      \n`? Might be nice to comment on why such an odd string with a specific number of spaces.

File: Gruntfile.js [hunk 2]
```diff
@@ -140,28 +144,32 @@ module.exports = function (grunt) {
...
-                    // @todo versions should be fetched from backend/db/s3, not implemented yet
-                    frontmatter.versions[module.version] = frontmatter.updated;
+                    const moduleVersions = versions[index];
+                    frontmatter.versions = Object.keys(moduleVersions).reduce((reducer, version) => {
+                        reducer[version] = {date: getFormattedDate(new Date(moduleVersions[version].timestamp)), latest: version == module.version};
+                        return reducer;
+                    }, {});
+
+                    (await processVersions(module.version, moduleVersions, Object.assign({}, frontmatter))).forEach(item => {
+                        grunt.file.write(`./content/modules/${index}/${item.version}${extension}`, `${JSON.stringify(item.frontmatter, null, 2)}     \n${item.content}`);
```
Comment (craigcomstock) [inline Gruntfile.js:168]: Again, this extra space is a bit confusing. Maybe a note on why?

File: Gruntfile.js [hunk 3]
```diff
@@ -140,28 +144,32 @@ module.exports = function (grunt) {
...
+                    });
                 }
                 // frontmatters end
-                grunt.file.write(`./content/modules/${index}${extension}`, `${JSON.stringify(frontmatter, null, 2)}     \n${content}`);
+                grunt.file.write(`./content/modules/${index}/_index${extension}`, `${JSON.stringify(frontmatter, null, 2)}     \n${content}`);
```
Comment (craigcomstock) [inline Gruntfile.js:172]: Magic whitespace again?

**Review by craigcomstock — COMMENTED** [review]
Looks fine, just one question about magic whitespace? Maybe just needs a comment somewhere to explain.

**Review by olehermanse — APPROVED** [review]
LGTM, please address Craig's comments in replies or in a follow up PR. I will test on staging site.

**Comment by aleksandrychev** [comment]
@craigcomstock I will replace these magic whitespaces with a regex in a following PR

**Pull request merged**